sale_coupon_advanced/models/sale_order.py

```diff
@@ -86,12 +83,21 @@ def _filter_pricelist_programs(self, programs):
         return programs
 
     def _create_new_no_code_promo_reward_lines(self):
+        programs = self.with_context(
+            force_not_ordered_no_code_reward_programs=True
+        )._get_applicable_no_code_promo_program()
+        for program in programs.filtered(lambda r: r._is_program_forced()):
+            self._create_counter_line_for_reward_product(program)
         super()._create_new_no_code_promo_reward_lines()
-        program = self._get_applicable_no_code_promo_program().filtered(
-            lambda p: p.reward_type == "use_pricelist"
+        # Can there be only one such pricelist program?
```

Comment:
there may be few but this will probably break the code we should return only one program with a pricelist